tachyon/math/polynomials/univariate/two_adic_multiplicative_coset.h

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#ifndef TACHYON_MATH_POLYNOMIALS_UNIVARIATE_TWO_ADIC_MULTIPLICATIVE_COSET_H_
+#define TACHYON_MATH_POLYNOMIALS_UNIVARIATE_TWO_ADIC_MULTIPLICATIVE_COSET_H_
+
+#include <optional>
+#include <utility>
+#include <vector>
+#include <memory>
+
+#include "tachyon/base/bits.h"
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/base/optional.h"
+#include "tachyon/math/base/lagrange_selectors.h"
+#include "tachyon/math/matrix/matrix_types.h"
+#include "tachyon/math/polynomials/univariate/univariate_evaluation_domain.h"
+
+namespace tachyon::math {
+
+template <typename F,
+          size_t MaxDegree = (size_t{1} << F::Config::kTwoAdicity) - 1>
+class TwoAdicMultiplicativeCoset
```

Comment:
Since this depends on `LagrangeSelectors` which is related to AIR, why don't you create this kind of class like `GWCExtension`, which is an extension class of the `crypto::GWC`?